decorator.rb

```diff
@@ -0,0 +1,24 @@
+require_relative 'nameable'
+
+class Decorator < Nameable
+  def initialize(nameable)
+    @nameable = nameable
+    super()
+  end
+
+  def correct_name
+    @nameable.correct_name
+  end
+end
+
+class CapitalizeDecorator < Decorator
+  def correct_name
+    @nameable.correct_name.capitalize
+  end
+end
+
+class TrimmerDecorator < Decorator
+  def correct_name
+    @nameable.correct_name.slice(0, 10)
+  end
+end
```

Comment:
- [ ] For your code to be modular and readable, I strongly recommend you separate these two classes into separate files. So create a capitalize_decorator.rb and trimmer_decorator.rb file and inherit each of them from Decorator class accordingly 👍🏽